tests/foreman/ui_airgun/test_errata.py

```diff
@@ -66,33 +193,447 @@ def test_end_to_end(session):
+            # force the host to consume an RH product with adding a cdn repo.
+            RHELAnsibleEngineRepository(cdn=True),
+            YumRepository(url=CUSTOM_REPO_URL),
         ]
     )
     repos_collection.setup_content(org.id, lce.id, upload_manifest=True)
-    with VirtualMachine(distro=DISTRO_RHEL7) as client:
+    with VirtualMachine(distro=repos_collection.distro) as client:
         repos_collection.setup_virtual_machine(client)
-        assert client.subscribed
-        client.run('yum install -y {0}'.format(FAKE_1_CUSTOM_PACKAGE))
+        assert _install_client_package(client, FAKE_1_CUSTOM_PACKAGE, errata_applicability=True)
+        # Adding content view filter and content view filter rule to exclude errata for the
+        # installed package.
+        content_view = entities.ContentView(
+            id=repos_collection.setup_content_data['content_view']['id']).read()
+        cv_filter = entities.ErratumContentViewFilter(
+            content_view=content_view, inclusion=False).create()
+        errata = entities.Errata(content_view_version=content_view.version[-1]).search(
+            query=dict(search='errata_id="{0}"'.format(CUSTOM_REPO_ERRATA_ID)))[0]
+        entities.ContentViewFilterRule(content_view_filter=cv_filter, errata=errata).create()
+        content_view.publish()
+        content_view = content_view.read()
+        content_view_version = content_view.version[-1]
+        promote(content_view_version, lce.id)
         with session:
-            session.organization.select(org.name)
-            errata = session.errata.read(CUSTOM_REPO_ERRATA_ID)
-            assert errata['details'] == ERRATA_DETAILS
-            assert errata['packages'] == ERRATA_PACKAGES
-            assert (
-                errata['repositories']['table'][0]['Name'] ==
-                repos_collection.custom_repos_info[-1]['name']
-            )
-            assert (
-                errata['repositories']['table'][0]['Product'] ==
-                repos_collection.custom_product['name']
-            )
-            result = session.errata.install(
-                CUSTOM_REPO_ERRATA_ID, client.hostname)
-            assert result['result'] == 'success'
+            session.organization.select(org_name=org.name)
+            _set_setting_value(errata_status_installable, True)
+            expected_values = {
+                'Status': 'OK',
+                'Errata': 'All errata applied',
+                'Subscription': 'Fully entitled',
+            }
+            host_details_values = session.host.get_details(client.hostname)
+            actual_values = {
+                key: value
+                for key, value in host_details_values['properties']['properties_table'].items()
+                if key in expected_values
+            }
+            assert actual_values == expected_values
+            _set_setting_value(errata_status_installable, False)
```

Comment:
what was initial state of the settings? Why do we not revert it to initial state in case of failure or success of current test case?

Reply:
In the test we do not care what was the initial value but we care of setting the needed value for test scenario. The fixture errata_status_installable will take care of that initial value and will revert to that initial value in any case success or failure of the test.
```python

 @fixture
def errata_status_installable():
    """Fixture to allow restoring errata_status_installable setting after usage"""
    errata_status_installable = entities.Setting().search(
        query={'search': 'name="errata_status_installable"'})[0]
    original_value = errata_status_installable.value
    yield errata_status_installable
    _set_setting_value(errata_status_installable, original_value)


```